react/components/ExtensionPoint/ComponentLoader.tsx

```diff
@@ -60,16 +68,65 @@ const ComponentLoader: FunctionComponent<Props> = props => {
     hydration,
   } = props
 
+  /**
+   * Slot props should ALWAYS be PascalCased.
+   * It is OK to not include componentProps in the dependency array
+   * since there is currently no way for users to ADD or UPDATE slots via CMS.
+   * What this means is that the slots variable below only needs to be
+   * computed once during runtime, since we know that, even if componentProps
+   * is updated, the props that function as Slots will NOT change.
+   */
+  const slots = useMemo(() => {
+    if (!componentProps) {
+      return {}
+    }
+
+    const slotNames = Object.keys(componentProps).filter(
+      key => key[0] !== key[0].toLowerCase()
+    )
+    const resultingSlotsProps: Record<string, ReactNode> = {}
+
+    slotNames.forEach(
+      slotName =>
+        (resultingSlotsProps[slotName] = generateSlot({
+          treePath,
+          slotName,
+          slotValue: componentProps[slotName],
+          hydration,
+        }))
+    )
```

Comment:
If you're not going to use a `reduce`, a `for-of` can be a tiny bit more readable than a `forEach`
```suggestion
const resultingSlotsProps: Record<string, ReactNode> = {}

for (const slotName of slotNames) {
  resultingSlotsProps[slotName] = generateSlot({
    treePath,
    slotName,
    slotValue: componentProps[slotName],
    hydration,
  })
}
```